.travis.yml

```diff
@@ -0,0 +1,31 @@
+  - secure: "md8HXICzWoRCi5sk2aGgPP8947okH2Vlvr+5Ka67mVLd42JOtWx3EE13cvB9UqUVPLM/qreb6mFCb97xNI+5L4mr9pX+CllxANsWbNA3qwVGwxCUdXwXMokpm9KYUsvJG132PJv4fIzGDe4qpwb5kT6tH1zxcxdqXvMlqAwMP57oTHXGtMSwCpy6pzhILnOsf9BqiiS+jkT17yM9p5NEvxgtJBfFOqZO/iO8kyCqN6eaRtAlw8pxh1nplUiJdXwpU+ZKG7Dl5HHRDfyZenhgFWGctbsaNqfy2Aj9fKjCYznMJ47cA2h+36bprCKvRDQ4r67AeJr+joF9mubQBriVGokuS2kiCAmDaf70mwcm+GP08uxbUGY45DTELbyB+iPsz3mR8YKclir2JPTS6es5oE6wvQSSLLOquwku6mwGQQ5BAdu9CwECd3weYykGSz0FpbzLguP0D/Ad0lH1tdmrfADQvVUfDPr716QyWSA4X1Nd1oMKm6pjVh86Jfub2AM7UCTfopG3Uibjc7+u2yQZa2tIq376ssR4SHSzU+W3uAOOVbhPSsf1GyVDtt129qSATV8fZzCSZkRdZODGhwn98TmzVxKY1dPNWUwhQ1ioRqjnCwU70pWKzF34wsR2YnuODK8RvzCrDztwbJ5Lm+FuGFC9cagXwz8wQMHCNhdq1Ls="
+  - secure: "wznPpNu1ANjLfn8N/vOK0jxUR5uX49pbM7cqMto1HAEC6iOqdl9y0PfJxhKH0Um9bmGkoDN5RlToK4WIfBKiFU52bHwlA/k01ipvNPBM1G/aeZBflSdBIKa48Qa+L47PN6FJYLh5UKFSk7+CFVE0Au9ILEiQ/XT5hOEeDNbE82D/Ou2SB/Hca0LE6Gku/QTB7OUU7nzJLBRDWmCPdU0/ah6aOjy967qBY6YfcNPc5TG5Akm3v8qIqTby8MG9KArE8f5z4yEi2QEO3zZA+kZaKS9BHiNQ8U/sW607KHWR0FQtzoN1g0efkWrkv4uMGA0o5WxN2H29MVH/bYMaG0Fk70QZQ9+B+g48x1JxTaogaj43h5GK0pLV4Q+p3eBCqwnvKTzh+92YtqsBuCrSoFn4dzV+OBukAqDRLxHwfWv4AaqWOWTZdxFUAMOuKk2L70Bh6LoTjCb1S5Yx8lFb8A12gz8Nj7PhrLJSVd89MNq0pGWsUeq56MQ0Ugjw+WzTWq6Mm3LxuXIhpCnV8tDyGuO3emLmxz56lpLbdguCPX0OLIuT0VyHXahrqlo3syAP469ELV10xDwN+bpwNF8O8EwNIRBsqAV3aliObuYSelWOLJbtxVza2rh4qwpxq2td3terjhimxIKLM2ZmxRH0asfhYd3xUr+uD0LKLINRYBLrtBQ="
+  - secure: "b9mICmlbGgVv9kzOtiadOSx98i5xdQE9Z8QO8RaVhgAurXA6BDixJ2nMev9/Y3m5g+ZITFrf6cJk5VbFtuRLgQym8A99wgBior+RY8mebuxjVbydTh4wPMcZIBKJhjikXV7tPneCCTUCA8GPFsyMVY69A3XunLWEusoQaLXIQCQYVCWNgIzRl3fiCP74fMegQWZtp17qMk3L4TdEwHxdNgqqC9pBCGjyXvNUBB8MjllREQqxw9GbnZ2Sb5ulIZYMaiIwNb3MU6ytMpKoTVhx1KPVCisAUkbUPTlDq2a+ck8c8OTpmb9zzA2zhaWCvmL+610abh8gRuDJc/QvZXgFt1MtE/D1+9k/ZQUbARmz+24AQyoNRJRqXUUWs68AheJytXOQmE7J3ZZ5+cVffaWk1DtIP7N5Ltm/nCtwp9mDDQsrM3dcZeOs/Jy0DE09eAzp44E8IRhwImLC5b5E2HVPw1uzEHWtQQMpDTX+M4o0D2d01lse8JppkFFyUVFTw2D9FBDOpl0UQy46zTV6mwM3BnfTCP60LXKs6stDMVqHdyNU8Ss27+Sl+WqMDfJKDlxzAcBlvIkPKIBbjZDSLPoxQ45ZqhezHMSfKPD7cI62U5eK2pqmNZmntb7r5GkOwW+YWKJA243PZUuhopaKmb53uqhLZr3HKs5NwW1YRIRUVaQ="
+
+# As recommended in:
+# https://docs.travis-ci.com/user/languages/java/#Caching
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+
+before_install:
+  - ./ci/before_install.sh
+
+install:
+  - JAVA_HOME=$(jdk_switcher home openjdk8) ./gradlew classes testClasses
```

Comment:
Do you still need to set `JAVA_HOME` here?

Reply:
Good point. `jdk_switcher use` sets `JAVA_HOME` appropriately. Using that instead of `jdk_switcher home`.